LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/LnsMessageType.cs

```diff
@@ -22,4 +25,40 @@ internal enum LnsMessageType
 
         */
     }
+
+    internal static class LnsMessageTypeExtensions
+    {
+        internal static bool TryParseLnsMessageType(string s, out LnsMessageType? lnsMessageType)
+        {
+            lnsMessageType = s switch
+            {
+                "version" => LnsMessageType.Version,
+                "router_config" => LnsMessageType.RouterConfig,
+                "jreq" => LnsMessageType.JoinRequest,
+                "updf" => LnsMessageType.UplinkDataFrame,
+                "dntxed" => LnsMessageType.TransmitConfirmation,
+                "dnmsg" => LnsMessageType.DownlinkMessage,
+                _ => null
+            };
+            return lnsMessageType is not null;
+        }
+
+        internal static LnsMessageType ParseAndValidate(string s, LnsMessageType? expectedType) =>
+            TryParseLnsMessageType(s, out var parsedType)
+               ? expectedType is null ^ parsedType == expectedType
```

Comment:
The `^` is very hard to reason about. Seems simpler to just go with `||`:

```suggestion
               ? expectedType is null || parsedType == expectedType
```